src/Helpers/OidStripper.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Helpers;
+
+use SimPod\PhpSnmp\Exception\GeneralException;
+use SimPod\PhpSnmp\Transport\SnmpClient;
+use function array_key_first;
+use function Safe\substr;
+use function strlen;
+use function strrpos;
+
+final class OidStripper
+{
+    /**
+     * @param array<string, mixed> $leafOidData
+     *
+     * @return array<int, mixed>
+     *
+     * @template T
+     * @psalm-param array<string, T> $leafOidData
+     * @psalm-return array<int, T>
+     */
+    public static function stripParent(array $leafOidData) : array
+    {
+        $firstKey = array_key_first($leafOidData);
+        if ($firstKey === null) {
+            throw GeneralException::new('expected non-empty array');
```

Comment:
capital first letter? not sure how to approach this but never seen lower case in php

Reply:
it's just part of the error message in the end ... I'd keep it this way, but I'll let that up to you

Reply:
`Expected non-empty array, empty given` + drop `Unexpected error` part from the exception? I'd say exception implies that it's unexpected

Reply:
empty given is implicit enough from "expected non-empty array" imo

Reply:
not sure if clear enough what I have done wrong as a user 🤔  mb it's fine, not sure